src/reporting/ReportingWindow.sol

```diff
@@ -289,4 +291,22 @@ contract ReportingWindow is DelegationTarget, Typed, Initializable {
         numberOfReportsByMarket[_market] += 1;
         return true;
     }
+
+    function getWinningPayoutDistributionHashFromFork(Market _market) constant returns (bytes32) {
```

Comment:
One oddity about this example and perhaps others is that it is the Market itself calling this function. I believe using the `MarketExtensions` library in the market contract itself would increase the size by the library size. That means for such cases I'd still be using other contracts as proxies to call the `MarketExtensions` library methods within the Market contract.